python_modules/dagster/dagster/_core/definitions/dependency.py

```diff
@@ -485,20 +486,24 @@ def from_dict(cls, dict_repr: Mapping[str, Any]) -> "NodeHandle":
         return NodeHandle(name=dict_repr["name"], parent=parent)
 
 
-class NodeInputHandle(
-    NamedTuple("_NodeInputHandle", [("node_handle", NodeHandle), ("input_name", str)])
-):
+@dagster_model
```

Comment:
you mentioned in the other PR that these might not need runtime checking, which you can do with `checked=False`. Based on bench marking checking makes it go from ~700ns to 1us so only worth worrying about if these get constructed in hot paths

Reply:
Ah, perfect. Added that.